catalog/dags/common/storage/media.py

```diff
@@ -113,8 +116,9 @@ def clean_media_metadata(self, **media_data) -> dict | None:
         - add `provider`,
         - add default `category`, if available.
 
-        Raises an error if missing any of the required fields:
-        `license_info`, `foreign_identifier`, `foreign_landing_url`, or `url`.
+        Raises an error if missing any of the required fields: `license_info`,
+        `foreign_identifier`, `foreign_landing_url`, or `url`. Or if an extracted
+        media type does not match the media type of the class.
```

Comment:
I don't have a suggestion here, but this part of the sentence is confusing. "extracted media type" refers to what we extract from the "filetype" in the provider script, or from the URL extension, right? So, if, for example, the API returns `jpg` as `filetype` or if the URL ends with `.jpg` (when the API does not return `filetype`) for an audio item?

Reply:
From the URL extension! That's a good question. The comment was not clear, so I updated it with more details.